src/governance/VoltTimelockController.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {CoreRefV2} from "@voltprotocol/refs/CoreRefV2.sol";
+import {VoltRoles} from "@voltprotocol/core/VoltRoles.sol";
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+
+/// @title An override of the regular OZ governance/TimelockController to allow uniform
+/// access control in the Volt system based on roles defined in Core.
+/// @dev The roles and roles management from OZ access/AccessControl.sol are ignored, we
+/// chose not to fork TimelockController and just bypass its access control system, to
+/// introduce as few code changes as possible on top of OpenZeppelin's governance code.
+/// @author eswak
+contract VoltTimelockController is TimelockController, CoreRefV2 {
+    constructor(
+        address _core,
+        uint256 _minDelay
+    )
+        CoreRefV2(_core)
+        TimelockController(_minDelay, new address[](0), new address[](0))
+    {}
+
+    /// @dev override of OZ access/AccessControl.sol inherited by governance/TimelockController.sol
+    /// This will check roles with Core, and not with the storage mapping from AccessControl.sol
+    function hasRole(
+        bytes32 role,
+        address account
+    ) public view virtual override returns (bool) {
+        bytes32 roleToCheck = role;
+        // Remap governance/TimelockController.sol non-virtual public constants to Volt roles
+        if (role == PROPOSER_ROLE) {
+            roleToCheck = VoltRoles.TIMELOCK_PROPOSER;
+        }
+        if (role == EXECUTOR_ROLE) {
+            roleToCheck = VoltRoles.TIMELOCK_EXECUTOR;
+        }
+        if (role == CANCELLER_ROLE) {
+            roleToCheck = VoltRoles.TIMELOCK_CANCELLER;
+        }
```

Comment:
Why not directly copy the OZ roles into core and VoltRoles so this mapping isn't needed?

Reply:
they don't follow exactly the same naming scheme, but you're right it's a bit overkill, I'll remove the mapping